packages/neon_framework/lib/src/storage/sqlite_persistence.dart

```diff
@@ -0,0 +1,234 @@
+// ignore_for_file: cascade_invocations
+
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:built_collection/built_collection.dart';
+import 'package:flutter/foundation.dart';
+import 'package:logging/logging.dart';
+import 'package:meta/meta.dart' show internal;
+import 'package:neon_framework/src/storage/persistence.dart';
+import 'package:path/path.dart' as p;
+import 'package:path_provider/path_provider.dart';
+import 'package:sqflite_common_ffi/sqflite_ffi.dart';
+
+final _log = Logger('SQLitePersistence');
+
+/// An SQLite backed cached persistence for preferences.
+///
+/// There is only one cache backing all `SQLitePersistence` instances.
+/// Use the [prefix] to separate different storages.
+/// Keys within a storage must be unique.
+///
+/// The persistence must be initialized with by calling `SQLitePersistence().init()`
+/// and awaiting it's completion. If it has not yet initialized a `StateError`
+/// will be thrown.
+@internal
+final class SQLitePersistence extends CachedPersistence {
+  /// Creates a new sqlite persistence.
+  SQLitePersistence({this.prefix = ''});
+
+  /// The prefix of this persistence.
+  ///
+  /// Keys within it must be unique.
+  final String prefix;
+
+  @override
+  Map<String, Object> get cache => globalCache[prefix] ??= {};
+
+  @visibleForTesting
+  static final Map<String, Map<String, Object>> globalCache = {};
+
+  @visibleForTesting
+  static Database? database;
+
+  /// Initializes all persistences by setting up the backing SQLite database
+  /// and priming the global cache.
+  ///
+  /// This must be called and completed before any sqlite persistence.
+  static Future<void> init() async {
+    if (database != null) {
+      return;
+    }
+
+    final appDir = await getApplicationSupportDirectory();
+    database = await openDatabase(
+      p.join(appDir.path, 'preferences.db'),
+      version: 1,
+      onCreate: onCreate,
+    );
+
+    await getAll();
+  }
+
+  @visibleForTesting
+  static Future<void> onCreate(Database db, int version) async {
+    await db.execute('''
+CREATE TABLE "preferences" (
+	"prefix"	TEXT NOT NULL,
+	"key"	TEXT NOT NULL,
+	"value"	TEXT NOT NULL,
+	PRIMARY KEY("prefix","key"),
+	UNIQUE("key","prefix")
+);
+''');
+  }
+
+  static Database get _requireDatabase {
+    if (database == null) {
+      throw StateError(
+        'Persistence has not been set up yet. Please make sure SQLitePersistence.init() has been called before and completed.',
+      );
+    }
+
+    return database!;
+  }
+
+  @visibleForTesting
+  static Future<void> getAll() async {
+    try {
+      final results = await _requireDatabase.rawQuery('SELECT * FROM preferences');
```

Comment:
Specify the columns that should be selected. If we'd add a new column in the future but don't want to select it here, then it would still be selected and potentially slow down everything.